server/test/api/enterprises.js

```diff
@@ -0,0 +1,110 @@
+/*global describe, it, beforeEach, process*/
+
+var chai = require('chai');
+var chaiHttp = require('chai-http');
+var expect = chai.expect;
+chai.use(chaiHttp);
+
+// workaround for low node versions
+if (!global.Promise) {
+  var q = require('q');
+  chai.request.addPromises(q.Promise);
+}
+
+// environment variables - disable certificate errors
+process.env.NODE_TLS_REJECT_UNAUTHORIZED = 0;
+
+// base URL
+var url = 'https://localhost:8080';
+var user = { username : 'superuser', password : 'superuser', project: 1};
+
+/**
+* The /Enterprises API endpoint
+*
+* This test suite implements full CRUD on the /Enterprises HTTP API endpoint.
+*/
+describe('The /Enterprises API endpoint', function () {
+  var agent = chai.request.agent(url);
+  var NUMBER_OF_CASHBOXES = 2;
+  
+  var newEnterprises = {
+    name : 'newEnterprises',
+    abbr : 'newEnterprises', 
+    email : 'newEnterprises@test.org', 
+    po_box : 'newEnterprises', 
+    phone : '2016', 
+    location_id : 'bda70b4b-8143-47cf-a683-e4ea7ddd4cff', 
+    currency_id : [2]
+  };
+
+  var updateEnterprises = {
+    name : 'updateEnterprises',
+    abbr : 'updateEnterprises', 
+    email : 'newEnterprises@test.org', 
+    po_box : 'newEnterprises', 
+    phone : '00904940950932016', 
+    location_id : 'bda70b4b-8143-47cf-a683-e4ea7ddd4cff', 
+    currency_id : [2]
+  };
+
+  it('POST /ENTERPRISES will register a valid Enterprises', function () { 
+    return agent.post('/enterprises')
+      .send({ enterprises : newEnterprises })
+      .then(function (confirmation) { 
+        expect(confirmation).to.have.status(201);
+        updateEnterprises.id = confirmation.body.id  
+      })
+      .catch(handle);
+  });
+
+  it('GET /CURRENCY FOR ENTERPRISES returns a currencies list ', function () { 
+    var TOTAL_CURRENCIES = 2;
```

Comment:
It is good to define variables like this.  However, you should move this variable at the top of your `describe()` function so that we can change it easily in the future and it could be shared with future tests.